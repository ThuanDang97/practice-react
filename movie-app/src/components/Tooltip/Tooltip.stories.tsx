import { Story } from '@storybook/react';
import Tooltip from './Tooltip';
import { TProps } from './Tooltip';

export default {
  title: 'Components/Tooltip',
  component: Tooltip,
};

const Template: Story<TProps> = (args): JSX.Element => <Tooltip {...args}>Hover</Tooltip>;

export const Basic: Story<TProps> = Template.bind({});
Basic.args = {
  text: 'Tooltip',
};
