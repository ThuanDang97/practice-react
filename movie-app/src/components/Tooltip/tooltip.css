.tooltip .tooltip-text {
  margin-top: 5px;
  visibility: hidden;
  width: 140px;
  background-color: rgb(3, 37, 65);
  color: #fff;
  text-align: center;
  padding: 5px 10px;
  border-radius: 6px;
  position: absolute;
  z-index: 1;
}

.tooltip:hover .tooltip-text {
  visibility: visible;
}

.tooltip .tooltip-text::after {
  content: ' ';
  position: absolute;
  bottom: 100%;
  left: 14%;
  margin-left: -5px;
  border-width: 5px;
  border-style: solid;
  border-color: transparent transparent rgb(3, 37, 65) transparent;
}
