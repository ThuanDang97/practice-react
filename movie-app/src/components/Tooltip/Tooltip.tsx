import { ReactNode } from 'react';
import './tooltip.css';

export type TProps = {
  children: ReactNode;
  text: string;
};

const Tooltip = (props: TProps): JSX.Element => {
  const { text, children } = props;
  return (
    <div className="tooltip" data-testid="tooltip">
      <div>{children}</div>
      <span className="tooltip-text">{text}</span>
    </div>
  );
};

export default Tooltip;
