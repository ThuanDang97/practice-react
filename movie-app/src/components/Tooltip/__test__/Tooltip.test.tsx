import { render, screen } from '@testing-library/react';
import Tooltip from '../Tooltip';

describe('Test Tooltip component', () => {
  const component = (
    <Tooltip text="Element">
      <span>Element</span>
    </Tooltip>
  );
  test('Component Tooltip matches DOM Snapshot', () => {
    const { container } = render(component);

    expect(container).toMatchSnapshot();
  });

  test('Component Tooltip should render correctly', () => {
    render(component);

    expect(screen.getByTestId('tooltip')).toHaveTextContent('Element');
  });
});
