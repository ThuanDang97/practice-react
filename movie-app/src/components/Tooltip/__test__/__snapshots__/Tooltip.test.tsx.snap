// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Tooltip component Component Tooltip matches DOM Snapshot 1`] = `
<div>
  <div
    class="tooltip"
    data-testid="tooltip"
  >
    <div>
      <span>
        Element
      </span>
    </div>
    <span
      class="tooltip-text"
    >
      Element
    </span>
  </div>
</div>
`;
